common/static/common/js/components/views/paging_header.js

```diff
@@ -47,6 +47,7 @@
                 var selected = this.$('#paging-header-select option:selected');
                 this.sortOrder = selected.attr('value');
                 this.collection.setSortField(this.sortOrder);
+                this.collection.refresh();
```

Comment:
return this.collection.refresh();

Just in case anyone wants the promise, and to be consistent with below.